aten/src/ATen/native/ForeachUtils.h

```diff
@@ -87,19 +87,25 @@ bool has_same_attributes(Device expected_device, TensorList tensors) {
 bool will_promote_tensor(const Tensor& tensor, Scalar scalar) {
   // complex scalar + integral or boolean tensor will result in complex tensor
   if (scalar.isComplex() && at::isIntegralType(tensor.scalar_type(), /*includeBool*/ true)) {
-    return false;
+    return true;
+  }
+
+  // complex scalar + float tensor will result in complex tensor
+  if (scalar.isComplex() && at::isFloatingType(tensor.scalar_type())) {
+    return true;
```

Comment:
Can we reuse at::result_type and just do (pseudocode):
```
bool will_promote_tensor(const Tensor& tensor, Scalar scalar) {
    auto dtype = tensor.dtype();
    auto result_dtype = at::result_type(tensor, scalar);
    return dtype != result_dtype;
}
```
?
This makes it so that we don't need to maintain our separate implementation of this. I don't remember if there was some reason why we chose not to do this originally.

Reply:
Updated